### PR TITLE
Added function to get id from name

### PR DESCRIPTION
**Pull request by nature1996**
Added a way to access the ID using the location or item name. Added a documentation saying using those methods isn't recomended

### CODE REVIEW

File: apclient.hpp [hunk 1]
```diff
@@ -306,13 +306,33 @@ class APClient {
...
     }
 
+    /*Usage is not recomended*/
+    int64_t get_item_id(std::string name)
+    {
+        for (auto it : _items)
+        {
+            if (it.second == name) return it.first;
+        }
+        return -1;
```
Comment (black-sliver) [inline apclient.hpp:333]: Item -1 is Archipelago's "Nothing". If this is what you want to return that's fine, but maybe add a comment to clarify -1 is a valid item.

File: apclient.hpp [hunk 2]
```diff
@@ -306,13 +306,33 @@ class APClient {
...
     }
 
+    /*Usage is not recomended*/
+    int64_t get_location_id(std::string name)
+    {
+        for (auto it : _locations)
+        {
+            if (it.second == name) return it.first;
+        }
+        return -1;
```
Comment (black-sliver) [inline apclient.hpp:316]: Location -1 is Archipelago's "Cheat Console". Not sure that's ideal.
* -2 would be Archipelago's "Server"
* `std::numeric_limits<int64_t>::min()` would definitely be invalid, if that's the goal
  * requires `#include<limits>` on top of the file
  * could be added as a constant with `static constexpr INVALID_LOCATION = std::numeric_limits<int64_t>::min();` to make code more readable

**Review by black-sliver — CHANGES_REQUESTED** [review]

File: apclient.hpp (hunk 2 above)
Comment (nature1996) [reply apclient.hpp:317]: Actually, we could trow an error, so they have to use a try/catch

File: apclient.hpp (hunk 1 above)
Comment (nature1996) [reply apclient.hpp:334]: Could also do the same with error code

Comment (nature1996) [reply apclient.hpp]: Should be solved with commit [bae7be9](https://github.com/black-sliver/apclientpp/pull/5/commits/bae7be9d1c64d095812c1b4c70a2bc14b34df4f7)

File: apclient.hpp (hunk 2 above)
Comment (nature1996) [reply apclient.hpp]: Should be solved with commit [bae7be9](https://github.com/black-sliver/apclientpp/pull/5/commits/bae7be9d1c64d095812c1b4c70a2bc14b34df4f7)

**Comment by black-sliver** [comment]
So we don't forget: this could simplified using `...["location_name_to_id].value<int64_t>(name, INVALID_NAME_ID)` instead of writing that loop yourself (`.value` will do the loop under the hood); will merge as is to get that feature in. Just make a new PR when you want to clean it up.

**Pull request merged**